working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1863,52 +1863,108 @@ appears:
+there is only a single case, then the case body scope is that case's case scope
+and we're done. *If there is only a single case for the body, then any pattern
+variables used in the body have a single declaration in that case's pattern and
+need no other special handling.*
 
-When compiling the body shared by a set of cases with pattern variable sets `vs`
-(where default cases and labels have empty pattern variable sets), the enclosing
-scope is a synthesized scope `s` defined as:
+Otherwise, the case body scope `s` of a body used by a set of cases with pattern
+variable sets `vs` (where default cases and labels have empty pattern variable
+sets) is:
 
-1.  For each name in `n` appearing as a variable name in any of the pattern
+1.  Create a new empty scope `s` whose enclosing scope is the scope surrounding
+    the switch statement or expression. *Note that the body scope doesn't
+    include any of the case scopes as its enclosing scope. Instead, we
+    synthesize a scope that has new variables derived from the union of all of
+    the cases' variables.*
+
+2.  For each name `n` appearing as a variable name in any of the pattern
     variable sets in `vs`:
 
     1.  If `n` is defined in every pattern variable set in `vs` and has the same
         type and finality, then introduce `n` into `s` with that type and
-        finality.
+        finality. This is a *consistent variable* and is available for use in
+        the body.
 
         If any of the corresponding variables in `vs` are promoted, calculate
         the promoted type of the variable in `s` based on all of the promoted
         types of `n` in the cases in the same way that promotions are merged at
         join points.
 
-    2.  Else `n` is inconsistently defined by the cases. Introduce a variable
-        `n` into `s` (with arbitrary type and finality) and note that `n` is
-        an inconsistent variable.
+        *We declare a new variable because the enclosing scope of the body is
+        not any of the case scopes. The fact that this is a new variable and not
+        one of the variables declared by the cases is user-visible if a user
+        captures a case variable in a closure in the guard:*
+
+        ```dart
+        Function captured;
+
+        bool capture(Function closure) {
+          captured = closure;
+          return true;
+        }
+
+        switch (['before']) {
+          case [String a] when capture(() => print(a)):
+          case [_, String a]:
+            a = 'after';
+            captured();
+        }
+        ```
+
+        *This prints "before", not "after". In practice, users will rarely
+        notice this, the same way they rarely notice that an initializing formal
+        introduces a variable in the initializer list distinct from the
+        initialized field.*
+
+    2.  Else `n` is inconsistently defined by the cases. Introduce a new
+        variable `n` into `s` with arbitrary type and finality.
```

Comment:
I'd prefer "unspecified" to "arbitrary". It really has no type or finality.

That way, if any part of the specification mistakenly refers to the type or finality of an inconsistent variable, it cannot consistently continue.